source/Irrlicht/CMeshManipulator.cpp

```diff
@@ -1085,10 +1085,12 @@ ICPUMeshBuffer* CMeshManipulator::createMeshBufferWelded(ICPUMeshBuffer *inbuffe
 
 ICPUMeshBuffer* CMeshManipulator::createOptimizedMeshBuffer(const ICPUMeshBuffer* _inbuffer, const SErrorMetric* _requantErrMetric) const
 {
-	if (!_inbuffer || !_inbuffer->getMeshDataAndFormat())
-		return const_cast<ICPUMeshBuffer*>(_inbuffer);
+	if (!_inbuffer)
+		return NULL;
 
 	ICPUMeshBuffer* outbuffer = createMeshBufferDuplicate(_inbuffer);
+	if (!outbuffer->getMeshDataAndFormat())
```

Comment:
whats going on here?

why would createMeshBufferDuplicate give a null meshdataandformat if _inbuffer has a non null one ???

Reply:
But how you know that _inbuffer has a non null one?